src/nci/NfcTagManager.cpp

```diff
@@ -657,6 +711,28 @@ bool NfcTagManager::doNdefFormat()
   }
   sem_destroy(&sFormatSem);
 
+  if(isMifareTech(tag.mTechLibNfcTypes[0]) && !sFormatOk) {
+#ifdef NFCC_PN547
+    uint8_t key2[6] = {0xD3, 0xF7, 0xD3, 0xF7, 0xD3, 0xF7};
+    sem_init(&sFormatSem, 0, 0);
+
+    ALOGD("%s: Format with Second Key", __FUNCTION__);
+    status = EXTNS_MfcFormatTag(key2, sizeof(key2));
+    if (status == NFA_STATUS_OK) {
+      ALOGD("%s:2nd try wait for completion", __FUNCTION__);
+      sem_wait(&sFormatSem);
+      status = sFormatOk ? NFA_STATUS_OK : NFA_STATUS_FAILED;
+    } else {
+      ALOGE("%s: error status=%u", __FUNCTION__, status);
+      sem_destroy(&sFormatSem);
+    }
+
+    if(!sFormatOk) {
+      ALOGE("%s: Format with Second Key Failed", __FUNCTION__);
+    }
```

Comment:
|!sFormatOk| cannot merge to line 726, status return OK in line 721 doesn't mean format is success, the format result |sFormatOk| will be set by doFormatStatus which is called in callback function.